### PR TITLE
[Filebeat][Azure Module] Fixing event.outcome from result_type issue

### LINKED ISSUE #20990: Request to update the Azure Filebeat module pipeline for parsing Azure activity logs
**Describe the enhancement:**

The Azure Filebeat module is currently not parsing the field name `azure.activitylogs.properties.result` as `event.outcome`. Can this be updated please? I'm not that familiar with the Beats code, but I think the code below needs to be updated.

https://github.com/elastic/beats/blob/44e281f0985648e5cbc2a9e509178060736dd2ee/x-pack/filebeat/module/azure/activitylogs/ingest/pipeline.yml#L73-L77

I think the following code is where the Azure module is doing something similar for the audit logs:

https://github.com/elastic/beats/blob/7dec3275e5be9e872c7e6701fa125bbe7aadb352/x-pack/filebeat/module/azure/auditlogs/ingest/pipeline.yml#L34-L37

I've included a sample event below to help with the updates.

**Describe a specific use case for the enhancement or feature:**

This enhancement will affect the `event.outcome` field, which will be used in Elastic Security's prebuilt detection rules. The use of this field will ensure that our Azure detection rules align with the other rules that we have and the ECS schema.

**Example event:**

```{
  "_index": "filebeat-7.8.1-2020.07.30-000001",
  "_type": "_doc",
  "_id": "xK9U6XMBx-f0qx0mxQih",
  "_version": 1,
  "_score": null,
  "_source": {
    "agent": {
      "hostname": "azure",
      "name": "azure",
      "id": "8073c323-cf95-492e-a732-8c066c3844e3",
      "type": "filebeat",
      "ephemeral_id": "ee452fc8-d738-48ed-87ed-e74d74c442c3",
      "version": "7.8.1"
    },
    "log": {
      "level": "Informational"
    },
    "source": {
      "ip": "REDACTED"
    },
    "fileset": {
      "name": "activitylogs"
    },
    "azure-eventhub": {
      "sequence_number": 231,
      "consumer_group": "$Default",
      "offset": 8589981176,
      "eventhub": "insights-operational-logs",
      "enqueued_time": "2020-08-13T19:38:32.694Z"
    },
    "cloud": {
      "instance": {
        "name": "azure",
        "id": "701a74ef-bd09-4496-adc3-d838cc7fbf47"
      },
      "provider": "azure",
      "machine": {
        "type": "Standard_DS1_v2"
      },
      "region": "eastus"
    },
    "geo": {
      "continent_name": "North America",
      "region_iso_code": "REDACTED",
      "city_name": "REDACTED",
      "country_iso_code": "US",
      "region_name": "REDACTED",
      "location": {
        "lon": REDACTED,
        "lat": REDACTED
      }
    },
    "input": {
      "type": "azure-eventhub"
    },
    "@timestamp": "2020-08-13T19:35:41.399Z",
    "ecs": {
      "version": "1.5.0"
    },
    "service": {
      "type": "azure"
    },
    "host": {
      "hostname": "azure",
      "os": {
        "kernel": "5.3.0-1034-azure",
        "codename": "bionic",
        "name": "Ubuntu",
        "family": "debian",
        "version": "18.04.4 LTS (Bionic Beaver)",
        "platform": "ubuntu"
      },
      "containerized": false,
      "ip": [
        "REDACTED",
        "REDACTED"
      ],
      "name": "azure",
      "id": "0e9cc578aec046c7887befa87e43adf9",
      "mac": [
        "REDACTED"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "duration": 0,
      "module": "azure",
      "category": "Administrative",
      "dataset": "azure.activitylogs"
    },
    "azure": {
      "resource": {
        "provider": "Microsoft.aadiam",
        "id": "/tenants/2978caa4-d66c-4c7e-9e30-40034eb7c6f3/providers/Microsoft.aadiam"
      },
      "correlation_id": "9057fea2-b467-4922-824b-30efe1dc19a4",
      "activitylogs": {
        "operation_name": "Update policy",
        "operationVersion": "1.0",
        "tenantId": "2978caa4-d66c-4c7e-9e30-40034eb7c6f3",
        "category": "AuditLogs",
        "result_signature": "None",
        "properties": {
          "result": "success",
          "activityDisplayName": "Update policy",
          "resultReason": "",
          "correlationId": "9057fea2-b467-4922-824b-30efe1dc19a4",
          "loggedByService": "Core Directory",
          "operationType": "Update",
          "activityDateTime": "2020-08-13T19:35:41.3990077+00:00",
          "id": "Directory_9057fea2-b467-4922-824b-30efe1dc19a4_VMYVX_71392208",
          "additionalDetails": [],
          "category": "Policy",
          "targetResources": [
            {
              "administrativeUnits": [],
              "displayName": "Default Policy",
              "modifiedProperties": [
                {
                  "newValue": "\"\"",
                  "displayName": "Included Updated Properties",
                  "oldValue": null
                }
              ],
              "id": "c4b36e7c-6c8c-4a31-9dba-460621692286",
              "type": "Policy"
            }
          ],
          "initiatedBy": {
            "user": {
              "displayName": null,
              "roles": [],
              "ipAddress": "REDACTED",
              "id": "a350ac13-921f-46cd-b28e-a9b222d056f3",
              "userPrincipalName": "david@elasticbv.onmicrosoft.com"
            }
          }
        }
      }
    }
  },
  "fields": {
    "@timestamp": [
      "2020-08-13T19:35:41.399Z"
    ],
    "suricata.eve.timestamp": [
      "2020-08-13T19:35:41.399Z"
    ]
  },
  "highlight": {
    "azure.activitylogs.operation_name": [
      "@kibana-highlighted-field@Update policy@/kibana-highlighted-field@"
    ],
    "event.category": [
      "@kibana-highlighted-field@Administrative@/kibana-highlighted-field@"
    ],
    "event.module": [
      "@kibana-highlighted-field@azure@/kibana-highlighted-field@"
    ],
    "event.dataset": [
      "@kibana-highlighted-field@azure.activitylogs@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1597347341399
  ]
}
```

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

Adding a small fix to event.outcome from resulttype and adding a second property to event.outcome if result_type does not exist

## Why is it important?

Fixes small issues for event.outcome parsing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #20990

### CODE REVIEW

**Comment by P1llus** [comment]
@threat-punter If you could share a example doc before it was parsed as well that would be great, then I can add it as part of our test data :)

**Review by leehinman — APPROVED** [review]
LGTM

**Comment by threat-punter** [comment]
> @threat-punter If you could share a example doc before it was parsed as well that would be great, then I can add it as part of our test data :)

Unfortunately not, my Azure subscription expires and it looks like I don't have access to the raw event anymore.

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -555,6 +555,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Convert httpjson to v2 input {pull}20226[20226]
 - Improve Zeek x509 module with `x509` ECS mappings {pull}20867[20867]
 - Improve Zeek SSL module with `x509` ECS mappings {pull}20927[20927]
+- Added new properties field support for event.outcome in azure module {pull}20998[20998]
```
Comment (marc-gr) [inline CHANGELOG.next.asciidoc:558]: I think this needs to be reordered

**Comment by threat-punter** [comment]
@P1llus do you know if this one will get merged before the 7.10 feature freeze?

Once it's merged, I can go ahead and merge these detection rules ready for 7.10 too https://github.com/elastic/detection-rules/issues/143 and https://github.com/elastic/detection-rules/issues/129

**Comment by P1llus** [comment]
> @P1llus do you know if this one will get merged before the 7.10 feature freeze?
> 
> Once it's merged, I can go ahead and merge these detection rules ready for 7.10 too https://github.com/elastic/detection-rules/issues/143 and https://github.com/elastic/detection-rules/issues/129

This will be in 7.10 indeed, have just not had time to look at getting the merge.

@leehinman anything else needed before a merge?

**Comment by leehinman** [comment]
looks good.  I'm merge & open backport PR.

**Pull request merged**